lib/uri_parser.js

```diff
@@ -161,13 +162,146 @@ function parseQueryStringItemValue(value) {
   return value;
 }
 
+// Options that are known boolean types
+const BOOLEAN_OPTIONS = new Set([
+  'slaveok',
+  'slave_ok',
+  'sslvalidate',
+  'fsync',
+  'safe',
```

Comment:
`safe` is definitely not supported anymore, and I think the same is true for `fsync`. Also, I am surprised that we still support master/slave terminology.